app/models/piece.rb

```diff
@@ -82,4 +86,36 @@ def move_to!(new_row, new_column)
       end
     end
   end
+
+  def moving_into_check?(new_row, new_column)
+    #We will switch to the new coordinates temporarily and see if we are in check!
+    current_row = self.position_row
+    current_column = self.position_column
+    self.update_attributes(position_row: new_row, position_column: new_column)
+
+    #define my king and all the pieces from the other team
```

Comment:
So what I did was changed the return values of check?

Instead of returning true or false, we return the color of who is in check. If nobody is in check we return nil.

@SomeRandom42 I fixed all the check tests to reflect this change. I hope this doesn't mess you up with your checkmate method. It shouldn't be too big of a change and it simplified my method here.